level.go

```diff
@@ -38,6 +48,14 @@ func ParseLevel(s string) (Level, error) {
 	return 0, ErrLevelNotParsed
 }
 
+func (l Level) Formatted(withColour bool) string {
+	if withColour {
+		return l.ColouredString()
+	}
+
+	return l.String()
+}
+
```

Comment:
I'd not add this logic here, instead, I'd add this in the formatters.

What about passing a `type LevelFormatter func(Level) string` in the `TextConfig`/`JsonConfig` and providing an implementation?

```
type TextConfig struct {
	...
	LevelFormatter func(Level) string
}

// ...

switch t.Config.LevelFormatter {
	case nil:
		t.addElemQuoted(w, t.Config.LevelKeyName, t.Config.LevelFormatter(e.Level()))
	default:
		t.addElemQuoted(w, t.Config.LevelKeyName, e.Level().String())
}
```

Reply:
In this way users could also configure colors or other behavior they may  want to add

Reply:
Done, by the way adding the func as field in TextEncoder meant that in datadog_test.go a test failed because it was now not possible to compare TextConfigs due to the func field being there.